### ISSUE TITLE
.yml not compatible with macOS

### ISSUE THREAD

**Issue by FrickTobias** [FrickTobias/BLR#39]
## Problem
Trying to install dependencies doesn't seem to work on macOS.

```
conda env create -n blr -f ~/PycharmProjects/BLR/environment.yml 
Collecting package metadata: done
Solving environment: failed

ResolvePackageNotFound: 
  - tbb==2019.7=hc9558a2_0
  - llvm-openmp==8.0.0=hc9558a2_0
  - readline==7.0=h7b6447c_5
  - libcurl==7.64.1=h20c2e04_0
  - cryptography==2.7=py36h72c5cf5_0
  - bowtie2==2.3.5=py36he860b03_0
  - bcftools==1.9=ha228f0b_4
  - ncurses==6.1=he6710b0_1
  - sqlite==3.28.0=h7b6447c_0
  - curl==7.64.1=hbc83047_0
  - libstdcxx-ng==8.2.0=hdf63c60_1
  - libgcc-ng==8.2.0=hdf63c60_1
  - cutadapt==2.3=py36h14c3975_0
  - libssh2==1.8.2=h1ba5d50_0
  - bzip2==1.0.6=h14c3975_5
  - libffi==3.2.1=hd88cf55_4
  - pyrsistent==0.15.2=py36h516909a_0
  - pysam==0.15.2=py36h4b7d16d_3
  - cffi==1.12.3=py36h8022711_0
  - libdeflate==1.0=h14c3975_1
  - yaml==0.1.7=h14c3975_1001
  - wrapt==1.11.1=py36h516909a_0
  - samtools==1.9=h8571acd_11
  - cd-hit==4.8.1=hdbcaa40_2
  - pyyaml==5.1=py36h14c3975_0
  - psutil==5.6.2=py36h516909a_0
  - zlib==1.2.11=h7b6447c_3
  - xz==5.2.4=h14c3975_4
  - openjdk==8.0.152=h46b5887_1
  - tk==8.6.8=hbc83047_0
  - openssl==1.1.1b=h14c3975_1
  - pigz==2.4=h84994c4_0
  - krb5==1.16.1=h173b8e3_7
  - dnaio==0.3=py36h14c3975_1
  - perl==5.26.2=h14c3975_0
  - htslib==1.9=ha228f0b_7
  - libedit==3.1.20181209=hc058e9b_0
  - python==3.6.8=h0371630_0
  - datrie==0.7.1=py36h14c3975_0
```
## Similar problem
I found a [similar situation](https://stackoverflow.com/questions/49154899/resolvepackagenotfound-create-env-using-conda-and-yml-file-on-macos) where they move dependencies into a pip section. I'm not sure how the .yml file is generated but, could we implement a pip section in it and would it work with linux?

**Comment by FrickTobias** [comment]
@marcelm - Do you have any suggestions?

**Comment by marcelm** [comment]
I’m not sure what the problem is, but for now, you can specify the dependencies explicitly as `python=3.6 cutadapt bowtie2 pysam picard=2.10 cd-hit snakemake-minimal` instead of using `-f environment.yml`.

**Comment by marcelm** [comment]
I don’t think the advice about moving things to the pip section is a good one. We probably just need two different `environment.yml` files instead.

**Comment by marcelm** [comment]
Another possibility: When creating the file with `conda env export`, we could use the option `--no-builds`.

**Comment by FrickTobias** [comment]
I'm not sure of what the implications of the ```--no-builds``` flag would be but intuitively it makes sense. 

I'm thinking we maybe should (test the build on both operating systems)[https://docs.travis-ci.com/user/multi-os/] as well  but that is a separate issue or PR.

**Comment by marcelm** [comment]
Ok, good idea. I don’t use macOS myself, so it might be best if you set this up.

**Comment by johnne** [comment]
I'm pretty sure environment.yml file is made OS-specific when the builds are included. Ideally I think we should only include the main dependencies in the yml file, with version numbers but without build numbers.

**Comment by FrickTobias** [comment]
I added a check for osx in linux as a matrix as suggested by the [thread](https://docs.travis-ci.com/user/multi-os/) I posted earlier.

Then I used the ```--no-build``` to make ```.environment.yml``` compatible with both OS since #46 .

Lastly I added a check in ```.travis.yml``` for which OS it is running so it can:
- fetch the correct conda installation file #46 
- Install for osx md5sum (not included in stdlib) #47 

All commits in #46 where supposed to be in the osx branch.
